pkg/cli/config/config.go

```diff
@@ -351,3 +352,56 @@ func getRawKubeConfigLoader(kubeconfigPath string) clientcmd.ClientConfig {
 	// Return the clientcmd.ClientConfig (equivalent to ToRawKubeConfigLoader)
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
 }
+
+// ServerConnectionFlagsChanged returns true if any server connection flags are set.
+func ServerConnectionFlagsChanged(cmd *cobra.Command) bool {
+	return cmd.Flags().Changed("host") ||
+		cmd.Flags().Changed("token") ||
+		cmd.Flags().Changed("insecure-skip-tls-verify") ||
+		cmd.Flags().Changed("api-path")
+}
+
+// BuildDirectClientConfig builds a client.Config from CLI flags (host, token, api-path, insecure-skip-tls-verify).
+func BuildDirectClientConfig(p common.Params) (*client.Config, error) {
+	host := p.Host()
+	token := p.Token()
+	if host == "" || token == "" {
+		return nil, errors.New("--host and --token flag must be set if using direct connection flags")
+	}
+	rc := &rest.Config{
+		Host: host,
+	}
+	if p.APIPath() != "" {
+		rc.APIPath = p.APIPath()
+	}
+	if p.Token() != "" {
```

Comment:
[nit] This check is not strictly necessary IMO as we check that in line 368. Then we can simply assign that value to the BearerToken field when we initialize the config on line 372.